### PR TITLE
set buffer for scanner

### PR DESCRIPTION
**Pull request by gangtao**
adding buffer for scanner for large query result, now set the buffer to 1M

fix the issue of error `failed to query bufio.Scanner: token too long`

### CODE REVIEW

File: timeplus/client.go [hunk 1]
```diff
@@ -278,6 +278,8 @@ func (s *TimeplusClient) queryStreamV2(sql string, batchCount int, batchBufferTi
 	}
 
 	scanner := bufio.NewScanner(res.Body)
+	buf := make([]byte, 0, 64*1024)
+	scanner.Buffer(buf, 1024*1024)
```
Comment (zliang-min) [inline timeplus/client.go]: I don't feel like increasing the buffer size is the right way to fix the problem (actually it won't fix it for good). Have you tried [bufio.Reader](https://pkg.go.dev/bufio#Reader)? There is a `[ReadLine](https://pkg.go.dev/bufio#Reader.ReadLine)` method that allows you correctly read a line without worrying the buffer size (by checking the `isPrefix` returned value).

**Review by zliang-min — COMMENTED** [review]
There should be a better way to fix the problem.

File: timeplus/client.go (hunk 1 above)
Comment (gangtao) [reply timeplus/client.go]: yes, I found that, but handle `isPrefix` seems cause lots of complexity. it will need introduce extra logic to read one line

**Review by zliang-min — APPROVED** [review]

**Review by zliang-min — APPROVED** [review]

**Review by ye11ow — APPROVED** [review]

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**